apps/prairielearn/elements/pl-checkbox/pl-checkbox.py

```
@@ -725,7 +748,7 @@ def parse(element_html: str, data: pl.QuestionData) -> None:
     )
 
     # Check that the number of submitted answers is in the interval [min_options_to_select, max_options_to_select].
-    if not (min_options_to_select <= len(submitted_key) <= max_options_to_select):
+    if not (allow_blank and len(submitted_key) == 0 or min_options_to_select <= len(submitted_key) <= max_options_to_select):
```

Ruff (0.15.13)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)